File: MC_pi.py
from random import uniform

N = int(input('Number of trials: '))
M = 0

for i in range(N):
  rand_x = uniform(0,1)
  rand_y = uniform(0,1)
  r = rand_x ** 2 + rand_y ** 2
  if r <= 1:
    M += 1

print(f'For {N} trials, estimated value of pi is {4.0*M/N}.')

File: pi/MC_pi.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(){

  srand(time(NULL));
  double rand_x, rand_y, r, pi;
  int N;
  int M = 0;
  double max = RAND_MAX;

  printf("Number of trials:");
  scanf("%d", &N);
  
  for (int i=0; i<N; i++){
    rand_x = rand() / max;
    rand_y = rand() / max;
    r = rand_x * rand_x + rand_y * rand_y;
    if (r <= 1){
      M++;    
    }
  }

  pi = 4.0 * M / N;
  printf("For %d trials, estimated pi value is %lf\n", N, pi);
  return 0;

}

